File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
		int trans = 0;
		int count = 0;
		
		while(!s.equals("1")) {
            int origin_length = s.length();
			s = s.replaceAll("0", "");
			int new_length = s.length();
            count += origin_length - new_length;
			s = Integer.toBinaryString(new_length);
			trans++;
		}
		
		answer[0] = trans;
		answer[1] = count;
        return answer;
    }
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    trans = 0
    remove_zero = 0
    while True:
        origin_len = len(s)
        s = s.replace("0", "")
        remove_zero += origin_len - len(s)
        s = format(len(s), 'b')
        trans += 1
        if s == '1':
            break
    
    answer = [trans, remove_zero]
    
    return answer